cachix/src/Cachix/Client/PushQueue.hs

```diff
@@ -95,6 +96,7 @@ queryLoop workerState pushqueue pushParams = do
 exitOnceQueueIsEmpty :: IO () -> Async () -> Async () -> QueryWorkerState -> PushWorkerState -> IO ()
 exitOnceQueueIsEmpty stopProducerCallback pushWorker queryWorker queryWorkerState pushWorkerState = do
   putText "Stopped watching /nix/store and waiting for queue to empty ..."
+  Systemd.notifyStopping
```

Comment:
Is this the haskell counterpart for?

```
           If a service of Type=notify sends "EXTEND_TIMEOUT_USEC=...", this
           may cause the stop time to be extended beyond TimeoutStopSec=. The
           first receipt of this message must occur before TimeoutStopSec= is
           exceeded, and once the stop time has extended beyond
           TimeoutStopSec=, the service manager will allow the service to
           continue to stop, provided the service repeats
           "EXTEND_TIMEOUT_USEC=..."  within the interval specified, or
           terminates itself (see sd_notify(3)).
```

This is from `systemd.service(5)` manpage.

Reply:
Seems to be not implemented yet but https://hackage.haskell.org/package/systemd-2.3.0/docs/src/System.Systemd.Daemon.html#notifyStatus
allows to implement that. My haskell foo is not strong enough to come up with more concrete suggestions.

Reply:
But reading further below looks like this function would be the right place to extend the timeout.